bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/BulkTransformerHandler.java

```diff
@@ -0,0 +1,163 @@
+        if (!nquads.isEmpty()) {
+            persistNquads(nquads);
+        }
+        logger.info(LAST_CONSUMED_BATCH, batchResponse.contents().getFirst());
+        return null;
+    }
+
+    private void persistNquads(String nquads) {
+        var request = PutObjectRequest.builder()
+                          .bucket(ENVIRONMENT.readEnv(LOADER_BUCKET))
+                          .key(UUID.randomUUID() + NQUADS)
+                          .build();
+        s3OutputClient.putObject(request, RequestBody.fromString(nquads));
+    }
+
+    private String mapToNquads(IndexDocument content) {
+        // TODO: Refactor GraphName to allow construction of path of GraphName.
+        // This is a silly workaround to make GraphName work.
+        var documentIdentifier = content.getResource().at("/id").toString() + ".nt";
+        var graphName = GraphName.newBuilder()
+                            .withBase(ENVIRONMENT.readEnv("API_HOST"))
+                            .fromUnixPath(UnixPath.of(documentIdentifier))
+                            .build().toUri();
+        var body = content.getResource().toString();
+        return Nquads.transform(body, graphName).toString();
+    }
+
+    private static PutEventsRequestEntry constructRequestEntry(String lastEvaluatedKey, Context context,
+                                                               String location) {
+        return PutEventsRequestEntry.builder()
+                   .eventBusName(EVENT_BUS)
+                   .detail(new KeyBatchRequestEvent(lastEvaluatedKey, TOPIC, location).toJsonString())
+                   .detailType(MANDATORY_UNUSED_SUBTOPIC)
+                   .source(BulkTransformerHandler.class.getName())
+                   .resources(context.getInvokedFunctionArn())
+                   .time(Instant.now())
+                   .build();
+    }
+
+    private String extractContent(String key) {
+        var s3Driver = new S3Driver(s3BatchesClient, KEY_BATCHES_BUCKET);
+        logger.info(PROCESSING_BATCH_MESSAGE, key);
+        return attempt(() -> s3Driver.getFile(UnixPath.of(key))).orElseThrow();
+    }
+
+    private String getLocation(KeyBatchRequestEvent input) {
+        return nonNull(input) && nonNull(input.getLocation()) ? input.getLocation() : null;
+    }
+
+    private static String getStartMarker(KeyBatchRequestEvent input) {
+        return nonNull(input) && nonNull(input.getStartMarker()) ? input.getStartMarker() : null;
+    }
+
+    private ListObjectsV2Response fetchSingleBatch(String startMarker) {
+        return s3BatchesClient.listObjectsV2(
+            ListObjectsV2Request.builder().bucket(KEY_BATCHES_BUCKET).startAfter(startMarker).maxKeys(1).build());
+    }
+
+    private void sendEvent(PutEventsRequestEntry event) {
+        eventBridgeClient.putEvents(PutEventsRequest.builder().entries(event).build());
+    }
+
+    private List<IndexDocument> mapToIndexDocuments(String content, String location) {
```

Comment:
Should we embed the contexts here to avoid loading remote contexts? Can `DocumentUnwrapper` (from `data-loading`) be used here?

Reply:
Following PR